lib/Core/Pagerfanta.php

```diff
@@ -546,4 +556,14 @@ private function needsToIntegerConversion($value): bool
     {
         return (\is_string($value) || \is_float($value)) && (int) $value == $value;
     }
+
+    /**
+     * @param int $maxNbPages
+     */
+    public function setMaxNbPages($maxNbPages): self
+    {
+        $this->maxNbPages = $maxNbPages;
```

Comment:
There should be a guard in this method to make sure we have a valid number here.  It looks like based on the implementation, an exception should be thrown if someone gives a number less than 0 here.

Reply:
Ok.